library/core/src/iter/adapters/by_ref_sized.rs

```diff
@@ -40,3 +40,32 @@ impl<I: Iterator> Iterator for ByRefSized<'_, I> {
         self.0.try_fold(init, f)
     }
 }
+
+impl<I: DoubleEndedIterator> DoubleEndedIterator for ByRefSized<'_, I> {
+    fn next_back(&mut self) -> Option<Self::Item> {
```

Comment:
It's probably better to slap `#[inline]` on these methods to ensure that the whole iterator gets inlined when compiling with many CGUs.